game/tests.py

```
@@ -154,6 +168,66 @@ def test_email_failure_renders_error_and_removes_pending_user(self):
+
+    def setUp(self):
+        self.factory = RequestFactory()
+        self.user = User.objects.create_user(
+            username='pendingplayer',
+            email='pendingplayer@example.com',
+            password='StrongPass123!',
+        )
+        self.user.is_active = False
+        self.user.save()
+
+    def _build_request(self, *, last_otp_time):
```

Ruff (0.15.13)

[warning] Missing return type annotation for private function `_build_request`

(ANN202)